NSURLConnectionTests::RequestWithURL_MultiRunLoop is failing

Seeing this on debug and release.

Looks like some timing issues.

> 2017-05-25T23:32:38.5804507Z StartGroup: NSURLConnectionTests::RequestWithURL_MultiRunLoop
2017-05-25T23:32:38.5804507Z Establishing download task with url https://httpbin.org/cookies/set?winobjc=awesome
2017-05-25T23:32:38.6430016Z ##[error]: Error: FAILED: We should have received 3 delegate calls by now! 
2017-05-25T23:32:38.6430016Z  Value of: [connectionTestHelper.delegateCallOrder count]
2017-05-25T23:32:38.6430016Z   Actual: 0
2017-05-25T23:32:38.6430016Z Expected: 3 [File: d:\Build\Agent7\_work\6\s\tests\functionaltests\Tests\NSURLConnectionTests.mm, Function: void _testRequestWithURL(NSURLConnectionTestHelper *__strong), Line: 411]
2017-05-25T23:32:38.6430016Z ##[error]: Error: Verify: Fail [File: d:\Build\Agent7\_work\6\s\tests\functionaltests\Tests\NSURLConnectionTests.mm, Function: void _testRequestWithURL(NSURLConnectionTestHelper *__strong), Line: 411]
2017-05-25T23:32:38.6430016Z ##[error]: EndGroup: NSURLConnectionTests::RequestWithURL_MultiRunLoop [Failed]
2017-05-25T23:32:38.7367035Z StartGroup: NSURLSessionTests::DownloadTaskWithURL_WithCompletionHandler
2017-05-25T23:32:38.7367035Z Establishing download task with url <NSCFURL: 0EFE8528>
2017-05-25T23:32:38.7367035Z ##[error]: Error: FAILED: Waiting for download timed out! 
2017-05-25T23:32:38.7367035Z  Value of: (downloadLocation || downloadResponse || downloadError) || [condition waitUntilDate:[NSDate dateWithTimeIntervalSinceNow:c_testDownloadTimeoutInSec]]
2017-05-25T23:32:38.7367035Z   Actual: false
2017-05-25T23:32:38.7367035Z Expected: true [File: d:\Build\Agent7\_work\6\s\tests\functionaltests\Tests\NSURLSession.mm, Function: void NSURLSessionTests::DownloadTaskWithURL_WithCompletionHandler(), Line: 637]
2017-05-25T23:32:38.7367035Z ##[error]: Error: Verify: Fail [File: d:\Build\Agent7\_work\6\s\tests\functionaltests\Tests\NSURLSession.mm, Function: void NSURLSessionTests::DownloadTaskWithURL_WithCompletionHandler(), Line: 637]
2017-05-25T23:32:38.7367035Z ##[error]: EndGroup: NSURLSessionTests::DownloadTaskWithURL_WithCompletionHandler [Failed]
2017-05-25T23:32:38.7367035Z 




2017-05-25T23:32:38.9242053Z ##[error]:     NSURLConnectionTests::RequestWithURL_MultiRunLoop [Failed]
2017-05-25T23:32:38.9242053Z ##[error]:     NSURLSessionTests::DownloadTaskWithURL [Failed]
2017-05-25T23:32:38.9242053Z ##[error]:     NSURLSessionTests::DownloadTaskWithURL_WithCompletionHandler [Failed]

--- a/tests/functionaltests/Tests/NSURLConnectionTests.mm
+++ b/tests/functionaltests/Tests/NSURLConnectionTests.mm
@@ -328,6 +328,15 @@ typedef NS_ENUM(NSInteger, NSURLConnectionDelegateType) {
 
 - (NSURLConnection*)createAndStartConnectionWithRequest:(NSURLRequest*)request {
     NSURLConnection* connection = [[NSURLConnection alloc] initWithRequest:request delegate:self startImmediately:NO];
+
+    for (size_t i = 0; i < 2; ++i) {
+        if (_loops.count != _threads.count) {
+            [NSThread sleepForTimeInterval:0.5];
+        }
+    }
+
+    ASSERT_EQ_MSG(_threads.count, _loops.count, "Run loops did not start in time.");
+
     for (NSRunLoop* runLoop : _loops) {
         [connection scheduleInRunLoop:runLoop forMode:NSDefaultRunLoopMode];
     }

